linalg/solvers.cpp

```diff
@@ -85,8 +93,95 @@ void IterativeSolver::SetPrintLevel(int print_lvl)
       {
          print_level = print_lvl;
       }
+      else // Suppress output.
+      {
+         print_level = -1;
+         print_options = PrintLevel();
+      }
+   }
+#endif
+}
+
+void IterativeSolver::SetPrintLevel(PrintLevel options)
+{
+   print_options = options;
+
+   int derived_print_level = GuessLegacyPrintLevel(options);
+
+#ifndef MFEM_USE_MPI
+   print_level = derived_print_level;
+#else
+   if (dot_prod_type == 0)
+   {
+      print_level = derived_print_level;
+   }
+   else
+   {
+      int rank;
+      MPI_Comm_rank(comm, &rank);
+      if (rank == 0)
+      {
+         print_level = derived_print_level;
+      }
+   }
+#endif
+}
+
+DISABLE_WARNING_POP
+
+IterativeSolver::PrintLevel IterativeSolver::FromLegacyPrintLevel(
+   int print_level)
+{
+#ifdef MFEM_USE_MPI
+   int rank = 0;
+   if (comm != MPI_COMM_NULL)
+   {
+      MPI_Comm_rank(comm, &rank);
    }
 #endif
+
+   switch (print_level)
+   {
+      case -1:
+         return PrintLevel();
+      case 0:
+         return PrintLevel().Errors().Warnings();
+      case 1:
+         return PrintLevel().Errors().Warnings().Iterations();
+      case 2:
+         return PrintLevel().Errors().Warnings().Summary();
+      case 3:
+         return PrintLevel().Errors().Warnings().Summary().FirstAndLast();
+      default:
+#ifdef MFEM_USE_MPI
+         if (rank == 0)
+#endif
+            MFEM_WARNING("Unknown print level " << print_level <<
+                         ". Defaulting to level 0.");
+
+         return PrintLevel().Errors().Warnings();
+   }
+}
+
+int IterativeSolver::GuessLegacyPrintLevel(PrintLevel print_options_)
```

Comment:
This can either be `static` or it can be a member function of `PrintLevel` (e.g. `int legacy_level = print_level.GuessLegacyLevel();`).